### PR TITLE
Bug 1276208: Upgrade nodejs in dockerfiles

### PR DESCRIPTION
**Pull request by pmac**
- Change the base image to quay.io/mozmar/base
- Upgrade prod node dependencies
- Switch to yuglify for js and css minification

(cherry picked from commit 0ab71c8ddf867c0e4727a8924c216daca3b717ba)

### CODE REVIEW

**Comment by pmac** [comment]
@jgmize r? again

**Comment by pmac** [comment]
`WIP` label just because there may be some jenkins work to make this happen. Josh and I will coordinate.

**Comment by pmac** [comment]
Hey @pmac, update the integration tests docker file with `python-setuptools`.

**Comment by pmac** [comment]
Docker files are all updated @jgmize.

File: docker/dockerfiles/bedrock_base [hunk 1]
```diff
@@ -11,11 +11,10 @@ CMD ["./docker/run-prod.sh"]
...
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends python2.7 libpython2.7 gettext build-essential libxml2-dev libxslt1.1 libxslt1-dev python-dev python-pip nodejs npm git postgresql-client libpq-dev
-RUN update-alternatives --install /usr/bin/node node /usr/bin/nodejs 10
-
-COPY ./bin/pipstrap.py /app/bin/pipstrap.py
-RUN ./bin/pipstrap.py
+    apt-get install -y --no-install-recommends \
+                    gettext build-essential python python-dev python-setuptools \
+                    libxml2-dev libxslt1.1 libxslt1-dev \
+                    nodejs-legacy npm git postgresql-client libpq-dev
```
Comment (jgmize) [inline docker/dockerfiles/bedrock_base]: The postgresql-client install will not work until the latest base image includes https://github.com/mozmar/docker-base/pull/1, and as I've mentioned before I don't think it really belongs in the base image anyway-- we can optionally include it in the dev image later.

**Comment by jgmize** [comment]
This doesn't just affect docker files, this also affects native local dev envs. @alexgibson would you mind testing this locally? I'd also be interested in hearing if you have any feedback on the switch to yuglify.

**Comment by alexgibson** [comment]
@jgmize - I've tested this locally and it seems to work well for me. It would be good to test the minification on a demo instance however (and run the integration tests against it?). Could you put this up on a demo? I'd also like to manually spot check out most common pages.

File: docker/dockerfiles/bedrock_base [hunk 2]
```diff
@@ -11,11 +11,10 @@ CMD ["./docker/run-prod.sh"]
...
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends python2.7 libpython2.7 gettext build-essential libxml2-dev libxslt1.1 libxslt1-dev python-dev python-pip nodejs npm git postgresql-client libpq-dev
-RUN update-alternatives --install /usr/bin/node node /usr/bin/nodejs 10
-
-COPY ./bin/pipstrap.py /app/bin/pipstrap.py
-RUN ./bin/pipstrap.py
+    apt-get install -y --no-install-recommends \
+                    gettext build-essential python python-dev python-setuptools \
+                    libxml2-dev libxslt1.1 libxslt1-dev \
+                    nodejs-legacy npm git libpq-dev
```
Comment (jgmize) [inline docker/dockerfiles/bedrock_base]: missing python-pip

File: docker/dockerfiles/bedrock_integration_tests [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
 WORKDIR /app
 # Run the tests
 CMD ["/app/bin/run-integration-tests.sh"]
 
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends build-essential python python-dev python-pip libxml2-dev libxslt1.1 libxslt1-dev zlib1g-dev
+    apt-get install -y --no-install-recommends \
+                    build-essential python python-dev python-pip python-setuptools \
+                    libxml2-dev libxslt1.1 libxslt1-dev zlib1g-dev
```
Comment (jgmize) [inline docker/dockerfiles/bedrock_integration_tests]: same here: missing python-pip

**Comment by jgmize** [comment]
Unfortunately this won't deploy to a demo as-is (see previous comments), and I tried rebasing my docker-dev image on this branch and then adding the missing python-pip deps but ran into what I think are separate issues blocking the deployment, and I'm going to need to pause and get some sleep. @pmac, after you add python-pip to the apt-get commands in the Dockerfiles I commented about earlier, will you please try pushing to a demo branch?

File: docker/dockerfiles/bedrock_integration_tests (hunk 3 above)
Comment (pmac) [inline docker/dockerfiles/bedrock_integration_tests]: nope, it's there on this one. good catch on the other though.

**Comment by pmac** [comment]
I've added the pip dep and pushed a demo branch (`demo__nodejs-upgrade`), but I'm pretty sure a demo push won't work. The demo process doesn't build the base image, and this relies on changes to the base image. What I can do is build the image locally and push it to a demo, but that's not very useful IMHO.

**Comment by alexgibson** [comment]
I took a look over some of our most high traffic pages on [@pmac's demo instance](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/) and all look as expected:

- [x] [Homepage (new)](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/)
- [x] [Homepage (old)](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/de/)
- [x] [Download page](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/firefox/new/)
- [x] [Firstrun](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/firefox/48.0/firstrun/)
- [x] [Plugincheck](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/plugincheck/)
- [x] [Channel](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/firefox/channel/)
- [x] [Developer](https://bedrock-demo-nodejs-upgrade.us-west.moz.works/en-US/firefox/developer/)

**Comment by alexgibson** [comment]
Should this PR also update the node version used in CircleCI? Looks like it still says "set node.js version to 0.12.0"

**Comment by jgmize** [comment]
argparse and the version of hashin currently in requirements/dev.txt are incompatible with newer versions of pip. please cherry-pick https://github.com/mozilla/bedrock/commit/389a21fc327cef501e790e01b688f153744c0ef7 into this branch.

**Comment by pmac** [comment]
I just removed them. Don't need em.

**Comment by jgmize** [comment]
@pmac mind rebasing this?

**Comment by jgmize** [comment]
I deployed this to https://bedrock-demo-nodejs-upgrade.us-west.moz.works/ via https://ci.us-west.moz.works/job/bedrock_demo_nodejs_upgrade/10/console which ran https://gist.github.com/jgmize/12379859b6f72b7661e588d4cbbbae3f and then ran https://ci.us-west.moz.works/job/bedrock_integration_tests_runner/8461/ against the deployed demo. @alexgibson would you mind taking a look at that same set of pages you listed in https://github.com/mozilla/bedrock/pull/4271#issuecomment-242093642?

**Comment by pmac** [comment]
rebased

**Pull request merged**